WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/CouponLineDetailsViewModel.swift

```diff
@@ -40,4 +54,38 @@ final class CouponLineDetailsViewModel: ObservableObject {
         let couponLine = OrderFactory.newOrderCouponLine(code: code)
         didSelectSave(couponLine)
     }
+
+    func validateAndSaveData(onCompletion: @escaping (Bool) -> Void) {
+        let action = CouponAction.validateCouponCode(code: code.lowercased(), siteID: siteID) { [weak self] result in
+            switch result {
+            case let .success(couponExistsRemotely):
+                if couponExistsRemotely {
+                    self?.saveData()
+                    onCompletion(true)
+                } else {
+                    self?.notice = Notice(title: Localization.couponNotFoundNoticeTitle,
+                                          feedbackType: .error)
+                    onCompletion(false)
+                }
```

Comment:
```suggestion
            case .success(true):
                self?.saveData()
                onCompletion(true)
            case .success(false):
                self?.notice = Notice(title: Localization.couponNotFoundNoticeTitle,
                                      feedbackType: .error)
                onCompletion(false)
```

also potentially the two failures could be combined? I'm not sure whether we need to show the difference between not found and not validated though...

Reply:
Good idea, done in 4bc3dd10ceecbe1daf658037094e074a3bcf7a20